tests/lib/rules/require-expect.js

```diff
@@ -294,6 +304,32 @@ ruleTester.run("require-expect", rule, {
             code: "test('name', function(assert) { other.expect(1); assert.ok(true); });",
             options: [],
             errors: [{ message: alwaysErrorMessage("assert.expect") }]
+        },
+
+        // "always" configuration - errors are reported on `test()` line
+        {
+            code: [
+                "module('some-module', function() {",
+                "    test('some-test', function(assert) {",
+                "        if (false) {",
+                "            assert.ok(true, 'needs expect');",
+                "        }",
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            options: [],
+            errors: [{
+                message: alwaysErrorMessage("assert.expect"),
+                type: "CallExpression",
+                line: 2
+            }]
+        },
+
+        // "always" configuration - multiple assert statements only report one error per test
+        {
+            code: "test('name', function(assert) { maybe(function() { assert.ok(true); assert.ok(true); }) });",
+            options: [],
+            errors: [{ message: alwaysErrorMessage("assert.expect") }]
```

Comment:
Can we check line and column here as well, so we know that the `test()` CallExpression is the one reported on? (Lots of CallExpression nodes in this code)